Example/Backpack/SwiftUI/Components/AppSearchModal/AppSearchModalExampleView.swift

```diff
@@ -0,0 +1,116 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2023 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+import Backpack_SwiftUI
+
+struct AppSearchModalExampleView: View {
+    
+    enum ResultsType {
+        case error
+        case loading
+        case content
+    }
+    
+    let resultsType: ResultsType
+    
+    var body: some View {
+        ZStack {
+            Color(BPKColor.canvasContrastColor)
+            
+            switch resultsType {
+            case .error:
+                makeAppSearchModal(with: .error(errorState))
+            case .loading:
+                makeAppSearchModal(with: .loading(loadingState))
+            case .content:
+                makeAppSearchModal(with: .content(contentState))
+            }
+        }
+    }
+    
+    private func makeAppSearchModal(with results: BPKAppSearchModalResults) -> some View {
+        BPKAppSearchModal(
+            title: "Search Modal",
+            inputText: .constant(""),
```

Comment:
nit: can you create a State for this instead of using a constant? so that when you try out the component in the screen it's more interactive

Reply:
Updated in [e2bba1b](https://github.com/Skyscanner/backpack-ios/pull/1820/commits/e2bba1b811539e52d90f842f9c6b16a0b1a33dd3)